lib/config/set-envs.js

```diff
@@ -0,0 +1,58 @@
+// Set environment variables for any non-default configs,
+// so that they're already there when we run lifecycle scripts.
+//
+// See https://github.com/npm/rfcs/pull/90
+
+const envName = name => `npm_config_${name.replace(/-/g, '_')}`
+
+const envKey = (key, val) => {
+  return !/^[\/@_]/.test(key) &&
```

Comment:
nit: regexp

Reply:
Given the above, does this just mean: String starting with `/@_`?

Reply:
We're filtering out all strings that start with slash, at, or underscore, yes.  Because those are potentially sensitive.